aip/general/0162.md

```diff
@@ -27,124 +27,108 @@ to the version of an API as a whole.
+resource revision type **should** inherit from the parent resource and has a field being the parent resource type.
 
 ```proto
-message Book {
-  // The name of the book.
-  string name = 1;
+message BookRevision {
+   // The name of the book revision.
+   string name = 1;
 
-  // Other fields…
-
-  // The revision ID of the book.
-  // A new revision is committed whenever the book is changed in any way.
-  // The format is an 8-character hexadecimal string.
-  string revision_id = 5 [
-    (google.api.field_behavior) = IMMUTABLE,
-    (google.api.field_behavior) = OUTPUT_ONLY];
+   // The snapshot of the book
+   Book book = 2 [(google.api.field_behavior) = IMMUTABLE];
 
   // The timestamp that the revision was created.
-  google.protobuf.Timestamp revision_create_time = 6
+  google.protobuf.Timestamp create_time = 3
     [(google.api.field_behavior) = OUTPUT_ONLY];
+
+  // tags set on the resource revisions by service or user.
+  repeated string tags = 4;
+  [(google.api.field_behavior) = OUTPUT_ONLY];
 }
 ```
 
-- The resource **must** contain a `revision_id` field, which **should** be a
-  string and contain a short, automatically-generated random string. A good
-  rule of thumb is the last eight characters of a [UUID version 4 (random)](UUID4).
-  - The `revision_id` field **must** document when new revisions are created
-    (see [committing revisions](#committing-revisions) below).
-  - The `revision_id` field **should** document the format of revision IDs.
-- The resource **must** contain a `revision_create_time` field, which
+- The resource revision **must** contain a field representing the snapshot of the parent resource.
+- The resource revision **must** contain a `create_time` field, which
   **should** be a `google.protobuf.Timestamp` (see [AIP-142][]).
-
-**Note:** A randomly generated string is preferred over other methods, such as
-an auto-incrementing integer, because there is often a need to delete or revert
-revisions, and a randomly generated string holds up better in those situations.
+- The resource revision **may** contain a repeated field `tags` to represent tags that was set on the resource revision.
 
 ### Referencing revisions
 
-When it is necessary to refer to a specific revision of a resource, APIs
-**must** use the following syntax: `{resource_name}@{revision_id}`. For
-example:
-
-    publishers/123/books/les-miserables@c7cfa2a8
-
-**Note:** The `@` character is selected because it is the only character
-permitted by [RFC 1738 §2.2][] for special meaning within a URI scheme that is
-not already used elsewhere.
-
-APIs **should** generally accept a resource reference at a particular revision
-in any place where they ordinarily accept the resource name. However, they
-**must not** accept a revision in situations that mutate the resource, and
-should error with `INVALID_ARGUMENT` if one is given.
+When referring to specific revision of a resource, APIs must use the following syntax
```

Comment:
I'd rewrite this section is "resource names for revisions".

There's already guidance on how to refer to other resources: https://google.aip.dev/122#fields-representing-another-resource. It's using the resource name.

So this should just suggest a resource name (`/foo/{}/revisions/{}`).

Reply:
I agree - I don't think this should be written in terms of "the following syntax". It should be "revisions ***must*** exist in a subcollection of the original resource, named `revisions`" and then the example.